### PR TITLE
PROC: add parent_PID

### PR DESCRIPTION
**Pull request by requilence**
See DEV-427:

...
Include the parent pid of each process.

 ```
{
    "PID": 24,
    "Name": "oom_reaper",
    "State": "sleeping",
    "Cmdline": "",
    "parent_PID": 0
}
```

### CODE REVIEW

File: processes_notwindows.go [hunk 1]
```diff
@@ -76,7 +76,13 @@ func processesFromProc() ([]ProcStat, error) {
 			log.Errorf("Failed to read cmdline(%s): %s", stats[0], err.Error())
 		}
 
-		stat := ProcStat{PID: pid, Name: string(bytes.TrimRight(comm, "\n")), Cmdline: strings.Replace(string(bytes.TrimRight(cmdline, "\x00")), "\x00", " ", -1)}
+		ppid, err := strconv.Atoi(string(stats[4]))
+
```
Comment (soupdiver) [inline processes_notwindows.go]: remove empty line

File: processes_notwindows.go [hunk 2]
```diff
@@ -76,7 +76,13 @@ func processesFromProc() ([]ProcStat, error) {
...
+		if err != nil {
+			log.Errorf("Failed to convert PPID(%s) to int: %s", stats[4], err.Error())
+		}
+
+		stat := ProcStat{PID: pid, ParentPID: ppid, Name: string(bytes.TrimRight(comm, "\n")), Cmdline: strings.Replace(string(bytes.TrimRight(cmdline, "\x00")), "\x00", " ", -1)}
```
Comment (soupdiver) [inline processes_notwindows.go]: To make it easier to read I would suggest to create `ProcState` in multiple lines and not one.

File: processes_notwindows.go [hunk 3]
```diff
@@ -161,12 +167,18 @@ func processesFromPS() ([]ProcStat, error) {
...
 
-		last := strings.Join(parts[2:], " ")
+		last := strings.Join(parts[3:], " ")
 		fileBaseWithArgs := filepath.Base(last)
 		fileBaseParts := strings.Fields(fileBaseWithArgs)
 
-		stat := ProcStat{PID: pid, Name: fileBaseParts[0], Cmdline: last}
-		switch parts[1][0] {
+		ppid, err := strconv.Atoi(string(parts[1]))
+
```
Comment (soupdiver) [inline processes_notwindows.go]: remove empty line

File: processes_notwindows.go (hunk 1 above)
Comment (requilence) [reply processes_notwindows.go]: Sometimes I put the empty line before the `if` block. Do you have a link to some convention on golang empty lines? I can't find any mention in the https://github.com/golang/go/wiki/CodeReviewComments

So then I will propose the PR to make all the empty lines consistent.

File: processes_notwindows.go [hunk 4]
```diff
@@ -113,6 +113,7 @@ func readProcFile(filename string) ([]byte, error) {
 	data, err := ioutil.ReadFile(filename)
 	if err != nil {
 		if os.IsNotExist(err) {
+			// if file doesn't exists
 			return nil, nil
```
Comment (soupdiver) [inline processes_notwindows.go]: Do we ignore the error in this case or do we have to return the `err`?

File: processes_notwindows.go (hunk 1 above)
Comment (soupdiver) [reply processes_notwindows.go]: There is not _everything_ mentioned in these guidelines.
I like to group code blocks logically. Function call together with error handling , declaration of variables and also `if` blocks.
Seems this comment was a mistake, the empty line is fine before the if block.
My bad.

File: processes_notwindows.go (hunk 4 above)
Comment (requilence) [reply processes_notwindows.go]: Will clarify this in another PR

Comment (soupdiver) [reply processes_notwindows.go]: hmmm not ideal but ok. I will open an issue an assign to you so we don't forget about this.

**Pull request merged**